src/mechanics/mesh/MeshGmsh.cpp

```diff
@@ -0,0 +1,378 @@
+    switch (gmshType)
+    {
+    case 1:
+        return mesh.CreateInterpolation(InterpolationTrussLinear(dimension));
+    case 2:
+        return mesh.CreateInterpolation(InterpolationTriangleLinear(dimension));
+    case 3:
+        return mesh.CreateInterpolation(InterpolationQuadLinear(dimension));
+    default:
+        throw NuTo::Exception(__PRETTY_FUNCTION__, "Unhandled gmsh element type.");
+    }
+}
+
+
+std::string GetPhysicalGroupName(const GmshFileContent& fileContent, int groupId)
+{
+    std::string physicalName{""};
+    for (const GmshPhysicalNames& gmshPhysicalName : fileContent.physicalNames)
+        if (gmshPhysicalName.id == groupId)
+        {
+            physicalName = gmshPhysicalName.name;
+            break;
+        }
+    return physicalName;
+}
+
+
+// Member functions
+// %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+NuTo::MeshGmsh::MeshGmsh(const std::string& fileName)
+{
+    ReadGmshFile(fileName);
+}
+
+const NuTo::Group<NuTo::ElementCollection>& NuTo::MeshGmsh::GetPhysicalGroup(std::string physicalName) const
+{
+    std::transform(physicalName.begin(), physicalName.end(), physicalName.begin(), ::toupper);
+    auto physGroupIt = mNamedPhysicalGroups.find(physicalName);
+    if (physGroupIt == mNamedPhysicalGroups.end())
+        throw Exception(__PRETTY_FUNCTION__, "Couldn't find group with name " + physicalName);
+    return *(physGroupIt->second);
+}
+
+const NuTo::Group<NuTo::ElementCollection>& NuTo::MeshGmsh::GetPhysicalGroup(int physicalGroupId) const
+{
+    auto physGroupIt = mPhysicalGroups.find(physicalGroupId);
+    if (physGroupIt == mPhysicalGroups.end())
+        throw Exception(__PRETTY_FUNCTION__, "Couldn't find group with ID " + std::to_string(physicalGroupId));
+    return physGroupIt->second;
+}
+
+std::vector<NuTo::NodeSimple*> NuTo::MeshGmsh::CreateNodes(const GmshFileContent& fileContent)
+{
+    std::vector<NodeSimple*> nodePtrVec(fileContent.maxNodeId - fileContent.minNodeId + 1, nullptr);
+    Eigen::VectorXd coords(fileContent.dimension);
+
+    for (const GmshNode& gmshNode : fileContent.nodes)
+    {
+        unsigned int vectorPos = gmshNode.id - fileContent.minNodeId;
+        for (int i = 0; i < fileContent.dimension; ++i)
+            coords[i] = gmshNode.coordinates[i];
+
+        if (vectorPos >= nodePtrVec.size())
+            throw Exception(__PRETTY_FUNCTION__, "Internal error - Calculated node position bigger than vector size!");
+        nodePtrVec[vectorPos] = &(mMesh.Nodes.Add(coords));
+    }
+    return nodePtrVec;
+}
+
+
+void NuTo::MeshGmsh::CreateElements(const GmshFileContent& fileContent,
+                                    const std::vector<NuTo::NodeSimple*>& nodePtrVec)
+{
+    std::map<int, const InterpolationSimple*> interpolationPtrMap;
+
+    for (GmshElement gmshElement : fileContent.elements)
+    {
+        // Skip node elements
+        if (gmshElement.type == 15)
+            continue;
```

Comment:
I think this is fine, for now, but I generally, could see a "node element" being useful. What does everyone else think?